### PR TITLE
[REVIEW] Adding CI

### PR DESCRIPTION
**Pull request by cjnolet**

### CODE REVIEW

**Comment by cjnolet** [comment]
I'm not sure how to pull in & build cuML here

**Comment by raydouglass** [comment]
> I'm not sure how to pull in & build cuML here

You shouldn't have to. Install it from conda if you need it to build. I recommend setting the channels to `-c rapidsai -c rapidsai-nightly` (or use labels for CUDA version if needed). This way it will pull either the current release or if your `meta.yml` depends on the next release, it will pull the nightly for that.

File: ci/cpu/build.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
+#!/bin/bash
+# Copyright (c) 2018, NVIDIA CORPORATION.
+######################################
+# cuML CPU conda build script for CI #
```
Comment (raydouglass) [inline ci/cpu/build.sh]: Fix the name

File: ci/cpu/build.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+logger "Build conda pkg for dask-cuml..."
+source ci/cpu/cuml/build_dask_cuml.sh
+
+################################################################################
+# UPLOAD - Conda packages
+################################################################################
+
+logger "Upload conda pkgs for libcuml..."
+source ci/cpu/libcuml/upload-anaconda.sh
```
Comment (raydouglass) [inline ci/cpu/build.sh]: The upload paths need to be fixed

**Comment by cjnolet** [comment]
Sounds good. I will do that. Thank you!

**Comment by cjnolet** [comment]
@dantegd @raydouglass Is there a way to run this in the github repository before merging it or should we merge this and update as we go?

File: ci/release/update-version.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+# Determine release type
+if [ "$RELEASE_TYPE" == "major" ]; then
+  NEXT_FULL_TAG="${NEXT_MAJOR}.0.0"
+  NEXT_SHORT_TAG="${NEXT_MAJOR}.0"
+elif [ "$RELEASE_TYPE" == "minor" ]; then
+  NEXT_FULL_TAG="${CURRENT_MAJOR}.${NEXT_MINOR}.0"
+  NEXT_SHORT_TAG="${CURRENT_MAJOR}.${NEXT_MINOR}"
+elif [ "$RELEASE_TYPE" == "patch" ]; then
+  NEXT_FULL_TAG="${CURRENT_MAJOR}.${CURRENT_MINOR}.${NEXT_PATCH}"
+  NEXT_SHORT_TAG="${CURRENT_MAJOR}.${NEXT_MINOR}"
```
Comment (raydouglass) [inline ci/release/update-version.sh]: This was a typo in cudf, should be `CURRENT_MINOR`

File: ci/gpu/build.sh [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+$CXX --version
+conda list
+
+################################################################################
+# BUILD - Build libcuml and cuML from source
+################################################################################
+
+logger "Install cuml"
+
+conda install -c rapidsai -c rapidsai-nightly -c conda-forge -c numba -c rapidsai/label/cuda${CUDA_REL} -c nvidia/label/cuda${CUDA_REL} -c pytorch -c defaults cuml dask distributed cudf dask_cudf
```
Comment (raydouglass) [inline ci/gpu/build.sh]: This project doesn't depend on CUDA right? So you should just use `-c rapidsai -c rapidsai-nightly`. You don't need the labels for the CUDA version.

File: ci/gpu/build.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+nvidia-smi
+
+logger "Check versions..."
+python --version
+$CC --version
+$CXX --version
+conda list
+
+################################################################################
+# BUILD - Build libcuml and cuML from source
```
Comment (raydouglass) [inline ci/gpu/build.sh]: Update the comments here and below L49.

File: conda/recipes/dask-cuml/meta.yaml [hunk 6]
```diff
@@ -34,7 +34,7 @@ requirements:
     - cudf
     - dask >=0.19.0
     - distributed >=1.23.0
-		- cuml
+		- dask-cuml
```
Comment (raydouglass) [inline conda/recipes/dask-cuml/meta.yaml]: I don't think this right. You shouldn't depend on yourself to run.

Also, looking at the full file, the build dependencies are really only on python? You probably need dask, cudf, etc to build right?

File: setup.py [hunk 7]
```diff
@@ -20,7 +20,7 @@
 install_requires = [
   'numpy',
   'dask',
-  'cuml',
+  'dask-cuml',
```
Comment (raydouglass) [inline setup.py]: Same as in meta.yml

**Review by raydouglass — CHANGES_REQUESTED** [review]

**Comment by raydouglass** [comment]
> @dantegd @raydouglass Is there a way to run this in the github repository before merging it or should we merge this and update as we go?

I'm hoping to get some CI on this repo tomorrow which can be used to test this.

File: ci/gpu/build.sh (hunk 4 above)
Comment (cjnolet) [reply ci/gpu/build.sh]: I'm confused by this. Do I need to use the pytorch and other conda channels in order to install cudf?

Comment (raydouglass) [reply ci/gpu/build.sh]: cudf needs (https://github.com/rapidsai/cudf#conda):
- nvidia
- rapidsai
- conda-forge
- defaults

I'm suggesting adding rapidsai-nightly AFTER rapidsai. That allows for picking up the nightlies if nightly dask-cuml depends on that.

**Comment by raydouglass** [comment]
rerun tests

**Review by raydouglass — APPROVED** [review]

**Pull request merged**